File: src/tree/common/quad-tree.ts
// This class definition comes from the problem itself and we cannot change it, or else our submission will not be
// accepted.
export class _Node {
  val: boolean;

  isLeaf: boolean;

  topLeft: _Node | null;

  topRight: _Node | null;

  bottomLeft: _Node | null;

  bottomRight: _Node | null;

  constructor(
    val?: boolean,
    isLeaf?: boolean,
    topLeft?: _Node,
    topRight?: _Node,
    bottomLeft?: _Node,
    bottomRight?: _Node
  ) {
    this.val = val === undefined ? false : val;
    this.isLeaf = isLeaf === undefined ? false : isLeaf;
    this.topLeft = topLeft === undefined ? null : topLeft;
    this.topRight = topRight === undefined ? null : topRight;
    this.bottomLeft = bottomLeft === undefined ? null : bottomLeft;
    this.bottomRight = bottomRight === undefined ? null : bottomRight;
  }
}

File: src/leetcode/tree/common/quad_tree.py
# This class definition comes from the problem itself and we cannot change it, or else our submission will not be
# accepted.
class Node:
    def __init__(
        self,
        val: bool = False,
        isLeaf: bool = False,
        topLeft: "Node | None" = None,
        topRight: "Node | None" = None,
        bottomLeft: "Node | None" = None,
        bottomRight: "Node | None" = None,
    ) -> None:
        self.val = val
        self.isLeaf = isLeaf
        self.topLeft = topLeft
        self.topRight = topRight
        self.bottomLeft = bottomLeft
        self.bottomRight = bottomRight
